File: Examples/VisualBasic/Working-Shapes/SetConnectorAppearance.vb
Imports Aspose.Diagram
Imports System
Public Class SetConnectorAppearance
    Public Shared Sub Run()
        ' ExStart:SetConnectorAppearance
        ' The path to the documents directory.
        Dim dataDir As String = RunExamples.GetDataDir_Shapes()

        ' Call a Diagram class constructor to load the VSDX diagram
        Dim diagram As New Diagram(dataDir & Convert.ToString("Drawing1.vsdx"))

        ' Get a particular page
        Dim page As Page = diagram.Pages.GetPage("Page-3")
        ' Get a dynamic connector type shape by id
        Dim shape As Shape = page.Shapes.GetShape(18)
        ' Set dynamic connector appearance
        shape.SetConnectorsType(ConnectorsTypeValue.StraightLines)

        ' Saving Visio diagram
        diagram.Save(dataDir & Convert.ToString("SetConnectorAppearance_out.vsdx"), SaveFileFormat.VSDX)
        ' ExEnd:SetConnectorAppearance
    End Sub
End Class

File: Examples/CSharp/Working-Shapes/SetConnectorAppearance.cs
using Aspose.Diagram;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Aspose.Diagram.Examples.CSharp.Working_Shapes
{
    public class SetConnectorAppearance
    {
        public static void Run()
        {
            // ExStart:SetConnectorAppearance
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_Shapes();

            // Call a Diagram class constructor to load the VSD diagram
            Diagram diagram = new Diagram(dataDir + "Drawing1.vsd");

            // Get a particular page
            Page page = diagram.Pages.GetPage("Page-3");
            // Get a dynamic connector type shape by id
            Shape shape = page.Shapes.GetShape(18);
            // Set dynamic connector appearance
            shape.SetConnectorsType(ConnectorsTypeValue.StraightLines);

            // Saving Visio diagram
            diagram.Save(dataDir + "SetConnectorAppearance_out.vsdx", SaveFileFormat.VSDX);
            // ExEnd:SetConnectorAppearance
        }
    }
}
